frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -3455,6 +3458,190 @@ def reinterpret(self, x: Conversion, *, bit_width: int) -> Conversion:
+            self.error(highlights)
+
+        chunk_size = self.configuration.multivariate_where_chunk_size
+        chunk_type = self.eint(chunk_size + 1)
+
+        if condition.bit_width != 1:
+            shifter = self.constant(self.i(condition.bit_width + 1), 2 ** (condition.bit_width - 1))
+            condition = self.mul(condition.type, condition, shifter)
+
+        condition = self.reinterpret(condition, bit_width=chunk_type.bit_width)
+
+        when_true_chunks = []
+        when_true_intermediate_type = self.tensor(
+            self.eint(resulting_type.bit_width),
+            shape=(np.zeros(condition.shape) + np.zeros(when_true.shape)).shape,
+        )
+
+        cursor = 0
+        while cursor < when_true.original_bit_width:
+            start = cursor
+            end = min(start + chunk_size, when_true.original_bit_width)
+            cursor += chunk_size
+
+            chunk = self.extract_bits(
+                self.tensor(chunk_type, shape=when_true.shape),
+                when_true,
+                slice(start, end),
+            )
+            packed_condition_and_chunk = self.add(
+                self.tensor(chunk_type, shape=when_true_intermediate_type.shape),
+                condition,
+                chunk,
+            )
+
+            when_true_chunks.append(
+                self.tlu(
+                    when_true_intermediate_type,
+                    packed_condition_and_chunk,
+                    [0 for _ in range(2**chunk_size)]
+                    + [x << start for x in range(2**chunk_size)],
+                )
+            )
+
+        when_false_chunks = []
+        when_false_intermediate_type = self.tensor(
+            self.eint(resulting_type.bit_width),
+            shape=(np.zeros(condition.shape) + np.zeros(when_false.shape)).shape,
+        )
+
+        cursor = 0
+        while cursor < when_false.original_bit_width:
+            start = cursor
+            end = min(start + chunk_size, when_false.original_bit_width)
+            cursor += chunk_size
+
+            chunk = self.extract_bits(
+                self.tensor(chunk_type, shape=when_false.shape),
+                when_false,
+                slice(start, end),
+            )
+            packed_condition_and_chunk = self.add(
+                self.tensor(chunk_type, shape=when_false_intermediate_type.shape),
+                condition,
+                chunk,
+            )
+
+            when_false_chunks.append(
+                self.tlu(
+                    when_false_intermediate_type,
+                    packed_condition_and_chunk,
+                    [x << start for x in range(2**chunk_size)]
+                    + [0 for _ in range(2**chunk_size)],
+                )
+            )
+
+        when_true_contribution = self.tree_add(when_true_intermediate_type, when_true_chunks)
+        when_false_contribution = self.tree_add(when_false_intermediate_type, when_false_chunks)
+
+        if when_true.is_signed:
```

Comment:
maybe an explanation in comment ?
or naming the constant -(2**when_true.original_bit_width) (extending_negative_bit ?)

Reply:
I'll add some comments.